proposals/lip-0054.md

```diff
@@ -221,50 +221,57 @@ def verify(trs: Transaction) -> None:
     if recoveryModule does not have a recover function:
         raise Exception("Module is not recoverable.")
 
+    keys = []
+
+    for entry in trsParams.storeEntries:
+        if entry.storeValue is EMPTY_BYTES:
+            raise Exception("Recovered store value cannot be empty.")
+
+        key = entry.substorePrefix + sha256(entry.storeKey)
+        keys.append(key)
+    
+    # Check that all keys are pariwise distinct, 
+    # meaning that we are not trying to recover the same entry twice.
+    if len(keys) != len(set(keys)):
```

Comment:
If we only check this, couldn't we run the same check on the params.storeEntries directly? I see no need to hash, for instance

Reply:
Right. I removed the hashing and I also construct the set directly without constructing the array before.